.github/workflows/on-release.yml

```diff
@@ -27,30 +27,18 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
-
+        with:
+          fetch-depth: 0 # fetch all history
       - name: Determine branch names
         run: |
           WITHOUT_V=${FULL_VERSION#v}
           PART_MAJOR=${WITHOUT_V%%.*}
           PART_MINOR=${WITHOUT_V#*.}
           MAJOR_MINOR=${PART_MAJOR}.${PART_MINOR}
           RELEASE_BRANCH="release/v${MAJOR_MINOR}"
-          WORKING_BRANCH_PREFIX="origin/tmp/${FULL_VERSION}"
+          WORKING_BRANCH_PREFIX="/tmp/${FULL_VERSION}"
           echo "RELEASE_BRANCH=${RELEASE_BRANCH}" | tee -a "$GITHUB_ENV"
           echo "WORKING_BRANCH_PREFIX=${WORKING_BRANCH_PREFIX}" | tee -a "$GITHUB_ENV"
-
-      - name: Find temporary branch
-        id: find-temporary-branch
-        run: |
-          working_branch_count=$(git branch --list -r "${WORKING_BRANCH_PREFIX}*" | wc -l)
-          if [[ "${working_branch_count}" -ne 1 ]]; then
-            echo "More than one (or zero) temporary branches found. Cannot finish transaction."
-            git branch --list "${WORKING_BRANCH_PREFIX}*"
-            exit 1
-          fi
```

Comment:
This isn't idempotent. We should cleanup the old branch before creating the new one instead.

We should also add a concurrency group to cancel ongoing runs of the release before we delete the old branch.

Reply:
IMHO this seems a bit overengineered. We would never run 2 concurrent releases would we? We can only retrigger after the existing one is finished.

Reply:
Did you write this step? Why is it relevant if there is tmp branch from another release? The release should only be concerned with its own release, or am I wrong?

Reply:
> IMHO this seems a bit overengineered. We would never run 2 concurrent releases would we? We can only retrigger after the existing one is finished.

I don't think it is. This are literally 3 lines.

Reply:
The problem is that here we need to find the correct tmp branch. So if there are multiple, it is possible we choose an old stale tmp branch. This code could be simplified by saying the tmp branch name must be `tmp/va.b.c` and not allow a suffix. Allowing multiple branches here is a recipe for disaster.

Reply:
> > IMHO this seems a bit overengineered. We would never run 2 concurrent releases would we? We can only retrigger after the existing one is finished.
> 
> I don't think it is. This are literally 3 lines.

did it, can you check @katexochen ?